airflow-core/src/airflow/api_fastapi/core_api/routes/public/task_instances.py

```
@@ -770,12 +772,16 @@ def post_clear_task_instances(
         )
 
     if not dry_run:
-        clear_task_instances(
-            task_instances,
-            session,
-            DagRunState.QUEUED if reset_dag_runs else False,
-            run_on_latest_version=body.run_on_latest_version,
-        )
+        try:
+            clear_task_instances(
+                task_instances,
+                session,
+                DagRunState.QUEUED if reset_dag_runs else False,
+                run_on_latest_version=body.run_on_latest_version,
+                prevent_running_task=body.prevent_running_task,
+            )
+        except ValueError as e:
+            raise HTTPException(status.HTTP_400_BAD_REQUEST, str(e)) from e
```

That pattern is ok but we should either:
- Check the 'message' for the ValueError. We only want this to handle `ValueError` caused by the `prevent_running_task` behavior. Other unexpected ValueError, shouldn't fall into 400 user errors.
- Change the exception raised there, maybe to raise a custom Exception such as `ClearRunningTaskException` or something like that, so you can catch it be sure not to interfer with other exceptions handling.